modules/diskq/tests/test_logqueue_disk.c

```diff
@@ -416,6 +418,74 @@ Test(logqueue_disk, restart_corrupted_with_multiple_queues)
   stop_grabbing_messages();
 }
 
+static inline void
+_pop_msgs(LogQueue *queue, gint num_of_messages)
+{
+  for (gint i = 1; i <= num_of_messages + 1; i++)
+    _pop_msg(queue);
+}
+
+Test(logqueue_disk, test_non_reliable_queue_front_cache_size)
+{
+  gint front_cache_size = 1024;
+  start_grabbing_messages();
+
+  const gchar *filename = "test_nrq_front_cache_size.qf";
+
+  DiskQueueOptions options = {0};
+  disk_queue_options_set_default_options(&options);
+  disk_queue_options_reliable_set(&options, FALSE);
+  disk_queue_options_capacity_bytes_set(&options, MIN_CAPACITY_BYTES);
+  disk_queue_options_front_cache_size_set(&options, front_cache_size);
+
+  StatsClusterKeyBuilder *driver_sck_builder = stats_cluster_key_builder_new();
+  StatsClusterKeyBuilder *queue_sck_builder = stats_cluster_key_builder_new();
+  LogQueue *queue = log_queue_disk_non_reliable_new(&options, filename, "test_nrq_front_cache_size",
+                                                  STATS_LEVEL0, driver_sck_builder, queue_sck_builder);
+  LogQueueDiskNonReliable *nrq = (LogQueueDiskNonReliable *) queue;
+  stats_cluster_key_builder_free(queue_sck_builder);
+  stats_cluster_key_builder_free(driver_sck_builder);
+
+  cr_assert(log_queue_disk_start(queue));
+  _assert_log_queue_disk_non_reliable_is_empty(queue);
+  cr_assert_eq(nrq->front_cache.limit, front_cache_size/2);
+  cr_assert_eq(nrq->front_cache_output.limit, front_cache_size/2);
+
+  LogPathOptions path_options = LOG_PATH_OPTIONS_INIT;
+  path_options.flow_control_requested = TRUE;
```

Comment:
I don't think this bit should be set here, more likely `ack_needed`?

Reply:
I seem to be wrong here, we do use flow_control_requested even in the queue to determine if we allow an overflow of the overflow window (flow controlled messages are never dropped, even if it means an overflow).

Reply:
`ack_needed` is already set by `LOG_PATH_OPTIONS_INIT`.